lib/client/index.js

```diff
@@ -0,0 +1,1710 @@
+        color = '#4cff00';
+      } else if (sgv < serverSettings.thresholds.bg_low) {
+        color = 'red';
+      } else if (sgv < serverSettings.thresholds.bg_target_bottom) {
+        color = 'yellow';
+      }
+    }
+
+    return color;
+  }
+
+  function sgvToColoredRange(sgv) {
+    var range = '';
+
+    if (browserSettings.theme === 'colors') {
+      if (sgv > serverSettings.thresholds.bg_high) {
+        range = 'urgent';
+      } else if (sgv > serverSettings.thresholds.bg_target_top) {
+        range = 'warning';
+      } else if (sgv >= serverSettings.thresholds.bg_target_bottom && sgv <= serverSettings.thresholds.bg_target_top) {
+        range = 'inrange';
+      } else if (sgv < serverSettings.thresholds.bg_low) {
+        range = 'urgent';
+      } else if (sgv < serverSettings.thresholds.bg_target_bottom) {
+        range = 'warning';
+      }
+    }
+
+    return range;
+  }
+
+
+  function generateAlarm(file, reason) {
+    alarmInProgress = true;
+    alarmMessage = reason && reason.title;
+    var selector = '.audio.alarms audio.' + file;
+
+    if (!alarmingNow()) {
+      d3.select(selector).each(function () {
+        var audio = this;
+        playAlarm(audio);
+        $(this).addClass('playing');
+      });
+    }
+
+    container.addClass('alarming').addClass(file === urgentAlarmSound ? 'urgent' : 'warning');
+
+    var skipPageTitle = isTimeAgoAlarmType(currentAlarmType);
+    updateTitle(skipPageTitle);
+  }
+
+  function playAlarm(audio) {
+    // ?mute=true disables alarms to testers.
+    if (querystring.mute !== 'true') {
+      audio.play();
+    } else {
+      showNotification('Alarm was muted (?mute=true)');
+    }
+  }
+
+  function stopAlarm(isClient, silenceTime) {
+    alarmInProgress = false;
+    alarmMessage = null;
+    container.removeClass('urgent warning');
+    d3.selectAll('audio.playing').each(function () {
+      var audio = this;
+      audio.pause();
+      $(this).removeClass('playing');
+    });
+
+    closeNotification();
+    container.removeClass('alarming');
+
+    updateTitle();
+
+    // only emit ack if client invoke by button press
+    if (isClient) {
+      if (isTimeAgoAlarmType(currentAlarmType)) {
+        container.removeClass('alarming-timeago');
+        var alarm = getClientAlarm(currentAlarmType);
+        alarm.lastAckTime = Date.now();
+        alarm.silenceTime = silenceTime;
+      }
+      socket.emit('ack', currentAlarmType || 'alarm', silenceTime);
+    }
+
+    brushed(false);
+  }
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+//draw a compact visualization of a treatment (carbs, insulin)
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+  function drawTreatment(treatment, scale, showValues) {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [This function's cyclomatic complexity is too high. (12)](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/799298422/issues/source?bid=2452379&fileBranchId=2725007#l1091)